libraries/rush-lib/src/logic/operations/PnpmSyncCopyOperationPlugin.ts

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+
+import { Async, FileSystem } from '@rushstack/node-core-library';
+
+import { OperationStatus } from './OperationStatus';
+import type { IOperationRunnerContext } from './IOperationRunner';
+import type { IPhasedCommandPlugin, PhasedCommandHooks } from '../../pluginFramework/PhasedCommandHooks';
+import type { OperationExecutionRecord } from './OperationExecutionRecord';
+import { pnpmSyncCopyAsync } from 'pnpm-sync-lib';
+
+const PLUGIN_NAME: 'PnpmSyncCopyOperationPlugin' = 'PnpmSyncCopyOperationPlugin';
+
+export class PnpmSyncCopyOperationPlugin implements IPhasedCommandPlugin {
+  public constructor() {}
+
+  public apply(hooks: PhasedCommandHooks): void {
+    hooks.afterExecuteOperation.tapPromise(
+      PLUGIN_NAME,
+      async (runnerContext: IOperationRunnerContext): Promise<void> => {
+        const record: OperationExecutionRecord = runnerContext as OperationExecutionRecord;
```

Comment:
Why is this cast necessary? What isn't exposed on the interface?

Reply:
Property `operation` does not exist on type `IOperationRunnerContext`.

Reply:
Can you add it to that interface? @dmichon-msft